### ISSUE TITLE
S1854 raised with a `try…catch…finaly` in a `foreach` with `continue`

### ISSUE THREAD

**Issue by thibaut-neoledge** [SonarSource/sonar-dotnet#2691]
### Description

The S1854 warning is raised if a variable is set in a block with a `continue` statement but only used in a `finally`

### Repro steps

```csharp
foreach (var item in new List<string>())
{
    bool warning = false;
    try
    {
        if (DateTime.UtcNow >= DateTime.Now)
        {
            warning = true;
            continue;
        }

        Debug.WriteLine("do something else.");
    }
    catch (Exception)
    {

        throw;
    }
    finally
    {
        // do something with the `warning` variable
        Debug.WriteLine(warning);
    }
}
```

### Expected behavior

S1854 should not be raised because the variable is used in the `finally` block.

### Actual behavior

S1854 is raised (because of the `continue` statement it seems).

### Related information

* SonarC# Version (SonarLint for VS) : 4.12.00
* Visual Studio Version : 2019 – 16.3.1

**Comment by andrei-epure-sonarsource** [comment]
thanks @thibaut-neoledge , it's a tricky problem well documented in #2550
I will close this as a duplicate
